parsl/executors/radical/executor.py

```
@@ -0,0 +1,447 @@
+                                    'pre_exec': self.pilot_env.get('pre_exec')
+                                    })
+
+        self.tmgr.add_pilots(pilot)
+        self.tmgr.register_callback(self.task_state_cb)
+
+        # create a bulking thread to run the actual task submittion
+        # to RP in bulks
+        if self.bulk_mode:
+            self._max_bulk_size = 1024
+            self._max_bulk_time = 3        # seconds
+            self._min_bulk_time = 0.1      # seconds
+
+            self._bulk_queue = queue.Queue()
+            self._bulk_thread = mt.Thread(target=self._bulk_collector)
+
+            self._bulk_thread.daemon = True
+            self._bulk_thread.start()
+
+        return True
+
+    def unwrap(self, func, args):
+        """
+        Unwrap a parsl app and its args for further processing.
+
+        Parameters
+        ----------
+        func : callable
+            The function to be unwrapped.
+
+        args : tuple
+            The arguments associated with the function.
+
+        Returns
+        -------
+        tuple
+            A tuple containing the unwrapped function, adjusted arguments,
+            and task type information.
+        """
+
+        task_type = ''
+
+        # ignore the resource dict from Parsl
+        new_args = list(args)
+        new_args.pop(0)
+        args = tuple(new_args)
+
+        while hasattr(func, '__wrapped__'):
+            func = func.__wrapped__
+
+        try:
+            if isinstance(func, partial):
+                try:
+                    task_type = inspect.getsource(func.args[0]).split('\n')[0]
+                    if BASH in task_type:
+                        task_type = BASH
+                        func = func.args[0]
+                    else:
+                        task_type = PYTHON
+
+                except Exception:
+                    logger.exception('unwrap failed')
+
+                    return func, args, task_type
```

maybe this should raise an exception (perhaps the existing exception) here rather than return?

---

> maybe this should raise an exception (perhaps the existing exception) here rather than return?

addressed here: https://github.com/Parsl/parsl/pull/2923/commits/9fe38da162aa0d2f6fb4981fb1bfc4f565cf3b30